app/src/main/java/com/napzak/market/data/detailpage/repositoryimpl/DetailPageRepositoryImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.napzak.market.data.detailpage.repositoryimpl
+
+import com.napzak.market.data.detailpage.datasource.DetailPageDataSource
+import com.napzak.market.data.detailpage.mapper.toDomain
+import com.napzak.market.domain.detailpage.repository.DetailPageRepository
+import com.napzak.market.domain.detailpage.model.ProductDetail
+import javax.inject.Inject
+
+class DetailPageRepositoryImpl @Inject constructor(
+    private val dataSource: DetailPageDataSource
+) : DetailPageRepository {
+    override suspend fun getProductDetail(productId: Long): Result<ProductDetail> = runCatching {
+        val response = dataSource.fetchProductDetail(productId)
+        if (response.status == 200) {
+            response.data.toDomain()
+        } else {
+            throw IllegalStateException("Unexpected status code: ${response.status}")
+        }
+    }
```

Comment:
p1: retrofit에선 header 내 상태 코드에 따라 error가 난 경우 표시를 해줍니다. 
위 코드처럼 직접 오류를 throw할 필요는 없어요! 
```suggestion
        val response = dataSource.fetchProductDetail(productId)
        response.data.toDomain()
```